var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -119,7 +120,7 @@ class Petsc(Package):
     depends_on('python@2.6:2.8', type='build')
 
     # Other dependencies
-    depends_on('boost', when='@:3.5+boost')
+    depends_on('boost', when='+boost')
```

Comment:
It looks like this was added in #2036 based on https://github.com/spack/spack/pull/1999#issuecomment-253293221. Ping @BarrySmith

Reply:
@alalazo there is also this comment, so I guess we can indeed keep the version constraint.

Reply:
Perhaps the following change is more appropriate
```
     # Other dependencies
-    depends_on('boost', when='@:3.5+boost')
     depends_on('metis@5:~int64+real64', when='@:3.7.99+metis~int64+double')
```
There was no direct dependency on boost even from old petsc versions.

There was some code in petsc which had this dependency - but this code was disabled by default. i.e - if petsc was configured with --with-sieve=1 - then it needed --with-boost=1. [sieve was never enabled in spack build of petsc - so boost was never required]

Now boost is primarily a depdendency for trilinos - so we would need --with-boost=1 for --with-trilinos=1 build. i.e for spack install petsc+trilinos

cc: @BarrySmith

Reply:
Also - perhaps the following change:
```
-    variant('boost',   default=True,  description='Activates support for Boost')
     variant('hypre',   default=True,
```

And somehow add

```
depends_on('boost', when='@:3.8+trilinos')
 options.append(with-boost=1) #in some if statement for petsc+trilinos
```